### PR TITLE
[ZEPL-3847] docs: Custom Image CLI

### PR DESCRIPTION
**Pull request by 1ambda**
image

image

image

image

image

### CODE REVIEW

**Comment by 1ambda** [comment]
/ping @sejunra Please help to review this one. I attached screenshots.

File: docs/guide/enterprise/custom_image_support.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+## Requirements
+
+Currently, CLI (zcr) **supports OSX and linux 64 bits.** Please contact us if you need support for additional OS.
+
+| Name  | Version | Description  |
+|---|---|---|
+| Docker | CE 18.06 +  | [Install Docker Community Edition](https://store.docker.com/search?type=edition&offering=community) |
+| AWS CLI  | 1.15.31 + |  [Install AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/installing.html) |
+
+Also, you need to AWS resources to push the built images into [AWS ECR](https://aws.amazon.com/ecr/)
```
Comment (sejunra) [inline docs/guide/enterprise/custom_image_support.md]: remove "to" - "Also, you need AWS resource..."

File: docs/guide/enterprise/custom_image_support.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+- [Create AWS key pair](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html#Using_CreateAccessKey) and export them into the terminal
+    ```bash
+    export IAM=1ambda AWS_ACCESS_KEY_ID={VALUE} AWS_SECRET_ACCESS_KEY={VALUE}
+    ```
+- The AWS Key should have the [AWS ECR push permission](https://docs.aws.amazon.com/AmazonECR/latest/userguide/ecr_managed_policies.html).
+- [Create AWS ECR](https://docs.aws.amazon.com/AmazonECR/latest/userguide/repository-create.html) in the same region where deployed ZEPL belongs to
+    ```bash
```
Comment (sejunra) [inline docs/guide/enterprise/custom_image_support.md]: "where deployed ZEPL belongs to"
to
"as the deployed ZEPL instance belongs"

File: docs/guide/enterprise/custom_image_support.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    # https://docs.aws.amazon.com/cli/latest/reference/ecr/create-repository.html
+    aws create-repository --repository-name {NAME} --region {REGION}
+    ```
+
+<br/>
+
+## Installation
+
+`zcr` can be installed / updated via the commands blow.
+
```
Comment (sejunra) [inline docs/guide/enterprise/custom_image_support.md]: change "blow" to "below"

File: docs/guide/enterprise/custom_image_support.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+<br/>
+
+## Usage
+
+`zcr` requires a definition file which defines custom packages for interpreters such as R libraries and Pip packages.
+With the `zcr create` command, users are able to build / push / register the custom interpreter image.
+
+```bash
+# make sure that you exported `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` as env variables.
+# replace `{PATH` with the your own definition file path
```
Comment (sejunra) [inline docs/guide/enterprise/custom_image_support.md]: change:
"replace `{PATH` with the your own definition file path"
to
"replace `{PATH}` with your own definition file path"

File: docs/guide/enterprise/custom_image_support.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+$ export TOKEN=$(aws ecr get-login --no-include-email --region us-west-2 | cut -d" " -f6)
+$ zic create --definition {PATH} --password ${TOKEN}
+
+# or can pass ECR auth token directly
+
```
Comment (sejunra) [inline docs/guide/enterprise/custom_image_support.md]: change to "or you can pass.."

File: docs/guide/enterprise/custom_image_support.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+For the definition file, please refer the section below.
+
+### Definition
+
+Definition of custom interpreter is written in YAML format. **These fields must be customized and set for each user.**
+
+- `meta.service_domain`: service domain including protocol (e.g. `https://...`)
+- `image.repository`: docker image repository (e.g. `0000000000.dkr.ecr.us-west-2.amazonaws.com`)
+- `image.name`: docker image name which should match with the ECR repository name (e.g. `custom-image`)
```
Comment (sejunra) [inline docs/guide/enterprise/custom_image_support.md:123]: if two ppl create an image with the same image.name, will the other be overwritten? If so, we should put a warning:

"If two users create images with the same image.name, the latter will overwrite the previous one."

**Review by sejunra — CHANGES_REQUESTED** [review]
just some spelling and grammar edits. 
looks good besides.

File: docs/guide/enterprise/custom_image_support.md (hunk 6 above)
Comment (1ambda) [reply docs/guide/enterprise/custom_image_support.md:123]: Added the description in `WARNING` 

image

**Comment by 1ambda** [comment]
image

image

image

image

image

**Comment by 1ambda** [comment]
/ping @sejunra Fixed the typo and updated screenshot and more description.

File: docs/guide/enterprise/custom_image_support.md [hunk 7]
```diff
@@ -90,9 +90,14 @@ $ zic create --definition ./spark2.definition.yaml \
   --password $(aws ecr get-login --no-include-email --region us-west-2 | cut -d" " -f6)
 ```
 
-You will see a message like if the whole process is successful.
+The commmand take few minutes depending on your network speed. At the end you will be able to see these kind of messages like if the whole process is successful.
```
Comment (sejunra) [inline docs/guide/enterprise/custom_image_support.md]: Replace:
"The commmand take few minutes depending on your network speed. At the end you will be able to see these kind of messages like if the whole process is successful."

with:

"The command will take a few minutes depending on your network speed. If the process is successful, you will see the following messages:"

**Review by sejunra — CHANGES_REQUESTED** [review]
some typo and grammar change.

**Comment by 1ambda** [comment]
@sejunra Fixed your last comment. Do you have additional comments?

**Review by sejunra — APPROVED** [review]

**Pull request merged**